### PR TITLE
add a "lagging" event to tell if the audio in playing is actually been paused because of insufficient buffer

### PR DESCRIPTION
**Pull request by garryyao**

### CODE REVIEW

File: widget.js [hunk 1]
```diff
@@ -61,8 +61,33 @@ define([
...
+			// actually paused caused by out of buffered bytes.
+			var slugging_timeout = 0;
+			var sluggish = 0;
+
 			me.on("audio5js/timeupdate", function (position, duration) {
-				var me = this;
+
+				// dismiss the slugging timeout
+				if (slugging_timeout) {
+					clearTimeout(slugging_timeout);
```
Comment (mikaelkaron) [inline widget.js]: could we refactor this to use promise.timeout instead?

**Comment by garryyao** [comment]
refactored with simplified with promise.delay

File: widget.js [hunk 2]
```diff
@@ -61,8 +64,39 @@ define([
...
-				var me = this;
+
+				cancelLaggingCheck();
+
+				// schedule a lagging check
+				if (me.player.playing) {
+					lagging = when.defer();
+					// the timeupdate even is fired up to 250ms interval
+					// http://dev.w3.org/html5/spec-preview/media-elements.html
+					lagging.promise = lagging.promise.timeout(500, 'lagging');
```
Comment (garryyao) [inline widget.js]: This 500 seems to suits well down to old IE7

File: bootstrap.js [hunk 3]
```diff
@@ -88,13 +87,6 @@ require({
...
 
-		jQuery(function ($) {
-			// Chrome allows you to click `forbidden` elements inside buttons
-			$(document.body).on("click", "button > span", function ($event) {
-				return $($event.target)
-					.parent()
-					.click();
-			});
-		});
+		jQuery(function ($) {});
```
Comment (mikaelkaron) [inline bootstrap.js]: ?

**Comment by mikaelkaron** [comment]
Looks good to me.

File: bootstrap.js (hunk 3 above)
Comment (garryyao) [inline bootstrap.js]: sorry a left over

**Pull request merged**